modules/caddypki/acmeserver/acmeserver.go

```diff
@@ -227,6 +259,62 @@ func (ash Handler) openDatabase() (*db.AuthDB, error) {
 	return database.(databaseCloser).DB, err
 }
 
+// makeClient creates an ACME client which will use a custom
+// resolver instead of net.DefaultResolver.
+func (ash Handler) makeClient() (acme.Client, error) {
+	for _, v := range ash.Resolvers {
+		addr, err := caddy.ParseNetworkAddress(v)
+		if err != nil {
+			// If a port wasn't specified for the resolver,
+			// try defaulting to 53 and parse again
+			if strings.Contains(err.Error(), "missing port in address") {
+				addr, err = caddy.ParseNetworkAddress(v + ":53")
+			}
+			if err != nil {
+				return nil, err
+			}
+		}
+		if addr.PortRangeSize() != 1 {
+			return nil, fmt.Errorf("resolver address must have exactly one address; cannot call %v", addr)
+		}
```

Comment:
```suggestion
		}
		if addr.StartPort == 0 {
				addr.StartPort = 53
		}
```

Reply:
That's already handled by `addr, err = caddy.ParseNetworkAddress(v + ":53")`

Reply:
Oh, I see what you mean. I think https://github.com/caddyserver/caddy/commit/1e18afb5c862d62be130d563785de5c58f08ae8e broke that port fallback. I'll need to dig deeper into this.

Reply:
Okay, I pushed another commit which adds a new `ParseNetworkAddressWithDefaults` function, so we can set the defaults to `udp` and port `53` for this case. Could you try this new build?